tests/ethpm/tools/test_builder.py

```diff
@@ -0,0 +1,268 @@
+import json
+from pathlib import Path
+
+from eth_utils.toolz import assoc
+import pytest
+
+from ethpm import Package
+from ethpm.backends.ipfs import get_ipfs_backend
+from ethpm.exceptions import ValidationError
+from ethpm.tools.builder import (
+    authors,
+    build,
+    contract_type,
+    description,
+    inline_source,
+    keywords,
+    license,
+    links,
+    manifest_version,
+    normalize_contract_type,
+    package_name,
+    pin_source,
+    return_package,
+    validate,
+    version,
+)
+
+BASE_MANIFEST = {"package_name": "package", "manifest_version": "2", "version": "1.0.0"}
+
+
+@pytest.fixture
+def owned_package(PACKAGING_EXAMPLES_DIR):
+    root = PACKAGING_EXAMPLES_DIR / "owned"
+    manifest = json.loads(Path(str(root / "1.0.0.json")).read_text())
+    compiler = json.loads(Path(str(root / "combined.json")).read_text())["contracts"]
+    return root, manifest, compiler
+
+
+@pytest.fixture
+def standard_token_package(PACKAGING_EXAMPLES_DIR):
+    root = PACKAGING_EXAMPLES_DIR / "standard-token"
+    manifest = json.loads(Path(str(root / "1.0.0.json")).read_text())
+    compiler = json.loads(Path(str(root / "combined.json")).read_text())["contracts"]
+    return root, manifest, compiler
+
+
+def test_builder_simple():
+    manifest = build(
+        {}, package_name("package"), manifest_version("2"), version("1.0.0"), validate()
+    )
+    assert manifest == BASE_MANIFEST
+
+
+def test_builder_simple_with_package(w3):
+    package = build(
+        {},
+        package_name("package"),
+        manifest_version("2"),
+        version("1.0.0"),
+        return_package(w3),
+    )
+    assert isinstance(package, Package)
+    assert package.version == "1.0.0"
+
+
+def test_builder_with_manifest_validation():
+    with pytest.raises(ValidationError):
```

Comment:
Maybe add a `match` kwarg to this to ensure the validation error contains the expected text (which I assume would include the string `"_invalid_package_name"`)

Reply:
Ooh, neat, didn't know about this. Seems like a pretty useful thing I should be using more often to make sure the correct error is being raised?